File: Sorting/Insertion_Sort/insert_sort.c
#include <stdio.h>
#include <conio.h>

static void printArray(int arr[],int s){
    for(int i = 0 ; i < s ; i++){
        printf("%d " ,arr[i]);
    }
    printf("\n");
}

static void insert_sort(int arr[] , int s){
    for(int i = 1 ; i < s ; i++ ){
        int cur = arr[i] ;
        int j = i - 1 ;

        while (j >= 0 && arr[j] > cur)
        {
            /* code */
            arr[j+1] = arr[j] ;
            j--;
        }

        arr[j+1] = cur ;
        
    }
       
}


int main(){
    printf("Enter the size of Array :-");
    int size ; 
    scanf("%d",&size);
    int arr[size];
    for(int i = 0 ; i < size ; i++){
        int t ;
        scanf("%d",&t);
        arr[i] = t ;
    }
    printf("THE INPUT ARRAY : - ");
    printArray(arr,size);
    insert_sort(arr,size);
    printf("THE OUTPUT ARRAY :- ");
    printArray(arr,size);
    return 0 ;
}

File: Sorting/Insertion_Sort/insertsort.java
package Sorting.Insertion_Sort;

import java.util.Scanner;

public class insertsort {

    public static void printArray(int[] arr){
        for(int i = 0 ; i < arr.length ; i++){
            System.out.print(arr[i]+" ");
        }
    }

    public static void insert_sorto(int[] arr){
        for(int i = 0 ; i < arr.length - 1 ; i++ ){
            int sm = i ;
            for(int j = i + 1 ; j < arr.length ; j++){
                if(arr[sm] > arr[j]){
                    sm = j ;
                }
                int t = arr[sm] ;
                arr[sm] = arr[i] ;
                arr[i] = t ;
            }
        }
    }

    public static void main(String[] args) {
        Scanner s = new Scanner(System.in);
        int size = s.nextInt();
        int[] arr = new int[size];
        System.out.println("The Location Of Array Variable In Meomary"+arr);
        for(int i = 0 ; i < size ; i++){
            arr[i] = s.nextInt();
        }
        System.out.println("THE INPUT ARRAY:- ");
        printArray(arr);
        insert_sorto(arr);
        System.out.println("THE OUTPUT ARRAY IS :-");
        printArray(arr);
    }
}
